conans/model/ref.py

```diff
@@ -129,17 +142,27 @@ def validate_string(value, reference_token=None):
             raise InvalidNameException(message)
 
     @staticmethod
-    def validate_name(name, version=False, reference_token=None):
+    def validate_name(name, reference_token=None):
         """Check for name compliance with pattern rules"""
         ConanName.validate_string(name, reference_token=reference_token)
         if name == "*":
             return
         if ConanName._validation_pattern.match(name) is None:
-            if version and ((name.startswith("[") and name.endswith("]")) or
-                            (name.startswith("(") and name.endswith(")"))):
-                return
             ConanName.invalid_name_message(name, reference_token=reference_token)
 
+    @staticmethod
+    def validate_version(name, version):
```

Comment:
```suggestion
    def validate_version(version, pkg_name):
```
Very confusing. Put the version as the first argument.

Reply:
Done